DataStructures/HashMap/Hashing/Open_Addressing.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Here I imnplement a hashing function
+ * the Open_Addressing object uses an array for hashing
+ * using quadratic hashing, we make it possible to add an element and retrieve its value from the array in constant time
+ */
+public class Open_Addressing {
+
+    public int tableSize; // number of SLOTS AVAILABLE
+    public int randomNumber; // the default random number
+    int index;
+    int hasher;
+    public int[] Table;
+
+    protected Open_Addressing(int index, int seed) {
+        this.index = index;
+        this.hasher = (int) (index - 1) / 2 + 1;
+        this.tableSize = power2(hasher);
```

Comment:
where's the method `power2()` ?

Reply:
I  had forgotten the function in my first commit. Now added